File: frontend/src/platform/types/auditLog.ts
export interface AuditLogUserOut {
  id: number
  name: string
  email: string
}

export interface AuditLogOut {
  id: number
  organization_id: number | null
  user_id: number | null
  user: AuditLogUserOut | null
  action: string
  resource_type: string | null
  resource_id: number | null
  ip_address: string | null
  details: Record<string, unknown> | null
  created_at: string
}

File: backend/src/platform/schemas/audit_log.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogUserOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    email: str


class AuditLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    organization_id: int | None
    user_id: int | None
    user: AuditLogUserOut | None
    action: str
    resource_type: str | None
    resource_id: int | None
    ip_address: str | None
    details: dict[str, Any] | None
    created_at: datetime
